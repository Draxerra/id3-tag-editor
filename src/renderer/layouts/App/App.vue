<template lang="pug">
    v-app#App
        v-toolbar(app)#AppToolbar
            v-toolbar-title#AppToolbarTitle ID3 Tag Editor
        v-content#AppContent
            v-container(fluid fill-height).pa-0#AppContainer
                router-view#AppView
        v-snackbar(:value="displaySnackbar"
                   @input="resetSnackbar"
                   :color="snackbar.color")#AppSnackbar {{ snackbar.message }}
            v-btn(flat
                  color="white"
                  @click="resetSnackbar")#AppSnackbarCloseBtn Close
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import { Getter, Mutation, State } from "vuex-class";

import { ISnackbar } from "~/src/types/snackbar";

import VApp from "vuetify/es5/components/VApp";
import VBtn from "vuetify/es5/components/VBtn";
import { VContainer, VContent } from "vuetify/es5/components/VGrid";
import VSnackbar from "vuetify/es5/components/VSnackbar";
import VToolbar, { VToolbarTitle } from "vuetify/es5/components/VToolbar";

@Component({
    components: {
        VApp,
        VBtn,
        VContainer,
        VContent,
        VSnackbar,
        VToolbar,
        VToolbarTitle,
    },
})
export default class App extends Vue {
    @Getter("snackbar/hasMessage") private displaySnackbar!: boolean;
    @Mutation("snackbar/reset") private resetSnackbar!: () => void;
    @State("snackbar") private snackbar!: ISnackbar;
}
</script>
