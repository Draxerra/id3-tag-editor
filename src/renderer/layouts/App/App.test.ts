import { config, createLocalVue, mount, RouterLinkStub } from "@vue/test-utils";
import App from "./App.vue";

import Vue from "vue";
import Vuetify from "vuetify/es5/components/Vuetify";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuetify);

const reset = jest.fn();
const snackbar = {
    getters: {
        hasMessage: () => true,
    },
    mutations: {
        reset,
    },
    namespaced: true,
    state: {
        color: "info",
        message: "Test Message",
    },
};
const store = new Vuex.Store({
    modules: {
        snackbar,
    },
});
const stubs = ["router-view"];

config.logModifiedComponents = false;

describe("App Layout", () => {

    it("is a Vue instance", () => {
        const wrapper = mount(App, { localVue, store, stubs });
        expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it("snapshot has not changed", () => {
        const wrapper = mount(App, { localVue, store, stubs });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("snackbar displays message", () => {
        const wrapper = mount(App, { localVue, store, stubs });
        expect(wrapper.find("#AppSnackbar").text()).toContain("Test Message");
    });

    it("snackbar does not appear if hasMessage getter is false", () => {
        const storeWithNoMessage = new Vuex.Store({
            modules: {
                snackbar: {...snackbar, getters: {
                    hasMessage: () => false,
                }},
            },
        });
        const wrapper = mount(App, { localVue, store: storeWithNoMessage, stubs });
        expect(wrapper.find("#AppSnackbar").exists()).toBeFalsy();
    });

    it("clicking snackbar close button fires reset fn", () => {
        const wrapper = mount(App, { localVue, store, stubs });
        expect(reset).not.toBeCalled();
        wrapper.find("#AppSnackbarCloseBtn").trigger("click");
        expect(reset).toBeCalled();
    });

});
