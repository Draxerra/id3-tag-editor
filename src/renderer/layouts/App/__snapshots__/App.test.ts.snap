// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Layout snapshot has not changed 1`] = `
<div data-app="true" class="application theme--light" id="App">
  <div class="application--wrap">
    <nav class="v-toolbar v-toolbar--fixed" style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);" id="AppToolbar">
      <div class="v-toolbar__content" style="height: 64px;">
        <div id="AppToolbarTitle" class="v-toolbar__title">ID3 Tag Editor</div>
      </div>
    </nav>
    <main class="v-content" style="padding: 64px 0px 0px 0px;" id="AppContent">
      <div class="v-content__wrap">
        <div class="container pa-0 fluid fill-height" id="AppContainer">
          <router-view-stub id="AppView"></router-view-stub>
        </div>
      </div>
    </main>
    <div class="v-snack v-snack--active v-snack--bottom" name="v-snack-transition" id="AppSnackbar">
      <div class="v-snack__wrapper info">
        <div class="v-snack__content">Test Message
          <button type="button" class="v-btn v-btn--flat white--text" id="AppSnackbarCloseBtn">
            <div class="v-btn__content">Close</div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
